src/execution/operator/aggregate/physical_streaming_window.cpp

```diff
@@ -420,15 +478,70 @@ OperatorResultType PhysicalStreamingWindow::Execute(ExecutionContext &context, D
 	for (idx_t col_idx = 0; col_idx < input.data.size(); col_idx++) {
 		chunk.data[col_idx].Reference(input.data[col_idx]);
 	}
-	// Compute window function
-	const idx_t count = input.size();
+	idx_t count = input.size();
+
+	//	Handle LEAD
+	if (state.lead_count > 0) {
+		//	Break the combined delayed + input into chunk + delayed
+		auto &delayed = state.delayed;
+		const idx_t available = delayed.size() + count;
+		//	If we don't have enough to produce a single row,
+		//	then just delay more rows and return nothing
+		if (available <= state.lead_count) {
+			delayed.Append(input, true);
```

Comment:
Is it necessary to set `resize` to `true` in the various appends here? We should be able to enforce that we are only ever keeping around `lead_count` values, no?